src/components/Dialogs/Conditions/AlphanumericCondition.js

```diff
@@ -0,0 +1,171 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    SingleSelectField,
+    SingleSelectOption,
+    Button,
+    Input,
+    Checkbox,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import classes from './styles/Condition.module.css'
+
+const NULL_VALUE = 'NV'
+export const OPERATOR_EQUAL = 'EQ'
+export const OPERATOR_NOT_EQUAL = '!EQ'
+export const OPERATOR_CONTAINS = 'LIKE'
+export const OPERATOR_NOT_CONTAINS = '!LIKE'
+export const OPERATOR_EMPTY = `EQ:${NULL_VALUE}`
+export const OPERATOR_NOT_EMPTY = `NE:${NULL_VALUE}`
+export const OPERATOR_EQUAL_CASE_SENSITIVE = 'IEQ'
+export const OPERATOR_NOT_EQUAL_CASE_SENSITIVE = '!IEQ'
+export const OPERATOR_CONTAINS_CASE_SENSITIVE = 'ILIKE'
+export const OPERATOR_NOT_CONTAINS_CASE_SENSITIVE = '!ILIKE'
+
+const operators = {
+    [OPERATOR_EQUAL]: i18n.t('exactly'),
+    [OPERATOR_NOT_EQUAL]: i18n.t('is not'),
+    [OPERATOR_CONTAINS]: i18n.t('contains'),
+    [OPERATOR_NOT_CONTAINS]: i18n.t('does not contain'),
+    [OPERATOR_EMPTY]: i18n.t('is empty / null'),
+    [OPERATOR_NOT_EMPTY]: i18n.t('is not empty / not null'),
+}
+
+const AlphanumericCondition = ({
+    condition,
+    onChange,
+    onRemove,
+    allowCaseSensitive,
+}) => {
+    let operator, value
+
+    if (condition.includes(NULL_VALUE)) {
+        operator = condition
+    } else {
+        const parts = condition.split(':')
+        operator = parts[0]
+        value = parts[1]
+    }
+
+    const setOperator = (input) => {
+        if (input.includes(NULL_VALUE)) {
+            onChange(`${input}`)
+        } else {
+            onChange(`${input}:${value || ''}`)
+        }
+    }
+
+    const setValue = (input) => {
+        onChange(`${operator}:${input || ''}`)
+    }
+
+    const toggleCaseSensitive = (state) => {
+        switch (operator) {
+            case OPERATOR_EQUAL:
+            case OPERATOR_EQUAL_CASE_SENSITIVE:
+                setOperator(
+                    state ? OPERATOR_EQUAL_CASE_SENSITIVE : OPERATOR_EQUAL
+                )
+                break
+            case OPERATOR_NOT_EQUAL:
+            case OPERATOR_NOT_EQUAL_CASE_SENSITIVE:
+                setOperator(
+                    state
+                        ? OPERATOR_NOT_EQUAL_CASE_SENSITIVE
+                        : OPERATOR_NOT_EQUAL
+                )
+                break
+            case OPERATOR_CONTAINS:
+            case OPERATOR_CONTAINS_CASE_SENSITIVE:
+                setOperator(
+                    state ? OPERATOR_CONTAINS_CASE_SENSITIVE : OPERATOR_CONTAINS
+                )
+                break
+            case OPERATOR_NOT_CONTAINS:
+            case OPERATOR_NOT_CONTAINS_CASE_SENSITIVE:
+                setOperator(
+                    state
+                        ? OPERATOR_NOT_CONTAINS_CASE_SENSITIVE
+                        : OPERATOR_NOT_CONTAINS
+                )
+                break
+        }
+    }
```

Comment:
I think this would work too. It's less readable, but shorter and would still work when new operators are added. Not sure sure how relevant that is though.
```suggestion
    const suffix = '_CASE_SENSITIVE'
    
    const toggleCaseSensitive = (isCaseSensitive) => {
        const newOperator = isCaseSensitive
            ? operator + suffix
            : operator.replace(suffix, '')

        if ([newOperator]) {
            setOperator(newOperator)
        }
    }
```

Reply:
Thanks for the suggestion. This got me thinking... If we're going down the `const suffix = '_CASE_SENSITIVE'` route, we might as well just treat the case sensitive part as a suffix (however as a prefix) in the const itself and just use the regular operators and append the ´I´ when needed. I'll see what I can put together...